chapter2/tests/t11/fastcgi_params

```diff
@@ -0,0 +1,27 @@
+
+fastcgi_param  QUERY_STRING       $query_string;
+fastcgi_param  REQUEST_METHOD     $request_method;
+fastcgi_param  CONTENT_TYPE       $content_type;
+fastcgi_param  CONTENT_LENGTH     $content_length;
+
+fastcgi_param  SCRIPT_NAME        $fastcgi_script_name;
+fastcgi_param  REQUEST_URI        $request_uri;
+fastcgi_param  DOCUMENT_URI       $document_uri;
+fastcgi_param  DOCUMENT_ROOT      $document_root;
+fastcgi_param  SERVER_PROTOCOL    $server_protocol;
+fastcgi_param  REQUEST_SCHEME     $scheme;
+fastcgi_param  HTTPS              $https if_not_empty;
+
+fastcgi_param  GATEWAY_INTERFACE  CGI/1.1;
+fastcgi_param  SERVER_SOFTWARE    nginx/$nginx_version;
+
+fastcgi_param  REMOTE_ADDR        $remote_addr;
+fastcgi_param  REMOTE_PORT        $remote_port;
+fastcgi_param  REMOTE_USER        $remote_user;
+fastcgi_param  SERVER_ADDR        $server_addr;
+fastcgi_param  SERVER_PORT        $server_port;
+fastcgi_param  SERVER_NAME        $server_name;
+fastcgi_param  SCRIPT_FILENAME    $document_root$fastcgi_script_name;
```

Comment:
prefer real path name, I have mentioned this in the lecture.

Reply:
> prefer real path name, I have mentioned this in the lecture.

@dmitry-mightydevops 
Like so?

Reply:
no, that way

```
fastcgi_param  SCRIPT_FILENAME    $realpath_root$fastcgi_script_name;
```

http://nginx.org/en/docs/http/ngx_http_core_module.html

**$realpath_root**
an absolute pathname corresponding to the [root](http://nginx.org/en/docs/http/ngx_http_core_module.html#root) or [alias](http://nginx.org/en/docs/http/ngx_http_core_module.html#alias) directive’s value for the current request, with all symbolic links resolved to real paths

**$document_root**
[root](http://nginx.org/en/docs/http/ngx_http_core_module.html#root) or [alias](http://nginx.org/en/docs/http/ngx_http_core_module.html#alias) directive’s value for the current request

Reply:
> no, that way
> 
> ```
> fastcgi_param  SCRIPT_FILENAME    $realpath_root$fastcgi_script_name;
> ```
> 
> http://nginx.org/en/docs/http/ngx_http_core_module.html
> 
> **$realpath_root** an absolute pathname corresponding to the [root](http://nginx.org/en/docs/http/ngx_http_core_module.html#root) or [alias](http://nginx.org/en/docs/http/ngx_http_core_module.html#alias) directive’s value for the current request, with all symbolic links resolved to real paths
> 
> **$document_root** [root](http://nginx.org/en/docs/http/ngx_http_core_module.html#root) or [alias](http://nginx.org/en/docs/http/ngx_http_core_module.html#alias) directive’s value for the current request

@dmitry-mightydevops 
Done